flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java

```diff
@@ -122,6 +136,32 @@ private void deployPerjob(YarnConfigOptions.UserJarInclusion userJarInclusion, J
 				assertThat(jobResult, is(notNullValue()));
 				assertThat(jobResult.getSerializedThrowable().isPresent(), is(false));
 
+				if (configuration.getInteger(YarnConfigOptions.FILE_REPLICATION) != -1) {
```

Comment:
Could you please add a new test case instead of modify current test cases? I think this pull request adds a new feature instead of code refactoring so that we should not touch any current test cases.

Reply:
+1

Reply:
Agree. I actually added two more test cases in YarnITCase. The change on this function is to minimize duplicated code.